src/miri.md

```diff
@@ -0,0 +1,47 @@
+# Miri
```

Comment:
This is a great start!

One thing I would find helpful is an example. Something like,

```rust
const MY_CONST: usize = 1 << 12;
let x = [0; MY_CONST]
```
would, I think, be sufficient. Maybe a walk through of evaluating the MIR to get the length of `x` would be good?

What do you think?

Reply:
Nit: `type Foo = [u8; MY_CONST];` may add less confusion, as nesting `const` inside a `fn` body can be misleading (even if it has no semantic effects).

Reply:
Heck yes to an example! =) Maybe use a `const fn` to explain?